pkg/sql/distsqlrun/processors.go

```diff
@@ -691,8 +691,10 @@ func (pb *ProcessorBase) moveToTrailingMeta() {
 	}
 
 	pb.State = stateTrailingMeta
-	if trace := getTraceData(pb.Ctx); trace != nil {
-		pb.trailingMeta = append(pb.trailingMeta, ProducerMetadata{TraceData: trace})
+	if pb.span != nil {
```

Comment:
Was this broken before?

Reply:
No, we just always called `StartInternal` which would create a span for us. If we don't do this check, we `getTraceData` from the span in `pb.Ctx` which is the parent span so we will send along a `RecordedSpan` that will also be created down the line, resulting in a `duplicate span` error.

Reply:
Ah, got it, cool